♻️ Refactor redis queue naming default value

Andy9822 opened the pull request:
We were using by default a queue named "queue" for the redis-based queues. Now we will defaultly name it using the Class and the mehod name to try to make it more unique in cases where the user doesn't really need to bother in passing a key.

## Example without passing a `key` value

![image](https://user-images.githubusercontent.com/19315023/178063106-21559afb-dd90-43dd-96eb-915bfd9f7220.png)

rafaeelaudibert commented on CHANGELOG.md:
```diff
@@ -1,5 +1,9 @@
 # CHANGELOG
 ​
+## v3.0.1
```
```suggestion
## v3.1.0
```

rafaeelaudibert reviewed (approved):
LGTM!

The changelog is a bit wrong

Andy9822 committed:
:recycle: Refactor logic for redis queue naming on each limited method

We were using by default a queue name "queue" for the redis-based queues. Now we will defaultly name it using the Class and the mehod name to try to make it more unique in cases where the user doesn't really need to bother in passing a key.

This way, the redis key is different for each class+method limited. For more specific cases, the user can still pass their own queue value as keyword argument.

 Also, we now support to send a `key` keyword argument so the limited method queue name can be customized.
